// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

syntax="proto2";
package ocean_kv;

option cc_generic_services = true;

message GetRequest {
      required string key = 1;
};

message GetResponse {
      required string value = 1;
};

message SetRequest {
      required string key = 1;
      required string value = 2;
};

message SetResponse {
      required uint32 code = 1;
};


service SimpleKVService {
      rpc Get(GetRequest) returns (GetResponse);
      rpc Set(SetRequest) returns (SetResponse);
};
